packages/pigeon/lib/java_generator.dart

```diff
@@ -180,16 +180,20 @@ void _writeHostApi(Indent indent, Api api) {
                 final bool isInt = arg.type.baseName == 'int';
                 final String argType =
                     isInt ? 'Number' : _javaTypeForDartType(arg.type);
-                final String argCast = isInt ? '.longValue()' : '';
                 final String argName = _getSafeArgumentName(index, arg);
+                final String argExpression = isInt
+                    ? '($argName == null) ? null : $argName.longValue()'
+                    : argName;
                 indent
                     .writeln('$argType $argName = ($argType)args.get($index);');
-                indent.write('if ($argName == null) ');
-                indent.scoped('{', '}', () {
-                  indent.writeln(
-                      'throw new NullPointerException("$argName unexpectedly null.");');
-                });
-                methodArgument.add('$argName$argCast');
+                if (!arg.type.isNullable && !arg.type.isVoid) {
```

Comment:
Oh, interesting question.  Can it be copypasta if I typed it in?  haha.  I got bit by this when I did return types and typed it in by reflex since its ambiguous if void is nullable or not.  Removed.